text/3873-build-std-context.md

```diff
@@ -0,0 +1,1237 @@
+  `no_std` crates ([crates.io#7306]).
+
+- `no_std` crates can accidentally and easily depend on crates that use `std`
+  which can result in build failures in some targets ([cargo#8798]).
+
+## Related work
+[related-work]: #related-work
+
+There are a variety of ongoing efforts, ideas, RFCs or draft notes describing
+features that are related or would be beneficial for build-std:
+
+- **[Opaque dependencies]**, [epage], May 2025
+  - Introduces the concept of an opaque dependency that has its own
+    `Cargo.lock`, `RUSTFLAGS` and `profile`
+  - Opaque dependencies could enable a variety of build-time performance
+      improvements:
+    - Caching - differences in dependency versions can cause unique instances of
+      every dependent crate
+    - Pre-built binaries - can leverage a pre-built artifact for a given opaque
+      dependency
+      - e.g. the standard library's distributed `rlib`s
+    - MIR-only/cross-crate lazy compilation - Small dependencies could be built
+      lazily and larger dependencies built once
+    - Optimising dependencies - dependencies could always be optimised when they
+      are unlikely to be needed during debugging
+
+# Motivation
+[motivation]: #motivation
+
+> [!IMPORTANT]
+>
+> This section lists all of the motivations that have been associated with
+> build-std in its various iterations, but not all of these use cases will be
+> addressed by this project goal.
+>
+> The motivations that will not be addressed are nevertheless mentioned here so
+> that reviewers have a more complete context for what has and hasn't been
+> desired of build-std over time.
+
+While the pre-built standard library has been sufficient for the majority of
+Rust users, there are a variety of use-cases which require the ability to
+rebuild the standard library.
+
+1. **Building the standard library without relying on unstable escape hatches**
+
+    - While tangential to the core of build-std as a feature, projects like Rust
+      for Linux want to be able to build crates from the standard library using
+      a stable toolchain without relying on escape hatches like
+      `RUSTC_BOOTSTRAP` that the Rust project does not encourage use of
+
+        - It is relatively straightforward to support this, hence its inclusion
+
+        - Cargo's implementation of build-std should be able to re-use whichever
+          mechanism is designed to address this
+
+2. **Building standard library crates that are not shipped for a target**
+
+    - Targets which have limited `std` support may wish to use the subsets of
+      the standard library which could work but are not shipped by the project
+      (e.g. `std` on `x86_64-unknown-none`)
+
+3. **Using the standard library with tier three targets**
+
+    - There is no stable mechanism for using the standard library on a tier
+      three target that does not ship a pre-built std
+
+    - While it is common for these targets to not support the standard library,
```

Comment:
I think you mean "std" here and not "the standard library", though it is quite common for tier 3 targets to support std. When scrolling through the list, most have an operating system and have had a working std build at some point. They do often break of course, something that build-std focused on tier 3 targets would likely need to address (which would be worth pointing out here I think).